src/services/api.tsx

```diff
@@ -19,55 +100,182 @@ const getAge = (birthDate: Date) => {
+                params += `&birthdate=gt${yyyy - 1}-${mm}-${dd}`;
+                return getPatients(params, 1000);
+              case ">":
+                params += `&birthdate=lt${yyyy}-${mm}-${dd}`;
+                return getPatients(params, 1000);
+              case "<":
+                params += `&birthdate=gt${yyyy}-${mm}-${dd}`;
+                return getPatients(params, 1000);
+            }
+            return {};
+          case "Diabète":
+            return getPatientPerCondition("73211009");
+          default:
+            console.info(`Paramètre ${x.label} non reconnu`);
+        }
+        return [];
+      })
+    )
   );
+
+  let finalBundle: Bundle = bundles[0];
+  bundles.map((bundle: Bundle) => {
+    if (finalBundle !== bundle && bundle.entry.length > 0) {
+      let listId = bundle.entry.map((x: any) => x.resource.id);
+      finalBundle.entry = finalBundle.entry.filter(
+        (entry: any) => listId.indexOf(entry.resource.id) >= 0
+      );
+    }
+    return false;
+  });
+  finalBundle.total = finalBundle.entry.length;
+
+  return addPatientsToBundle(finalBundle);
 };
 
-export const getPatientData = async (patientId: string) => {
+export const requestNextPatients = async (bundle: PatientBundle) => {
   /*
-    getPatientData requests for data from Patient resourcce of id patientId
-    return a Patient object.
-  */
-  let response = await client.search({
-    type: "Patient",
-    patient: patientId
+   * function requestNextPatients
+   * Return the same bundle with more patients, fetched from the nextLink attribute.
+   */
+  if (!bundle.nextLink) {
+    console.info("no link available");
+    return;
+  } else {
+    let newBundle: PatientBundle = (await makeRequestByURL(
+      bundle.nextLink
+    )) as PatientBundle;
+
+    newBundle.entry = bundle.entry.concat(newBundle.entry);
+
+    newBundle = addPatientsToBundle(newBundle);
+    newBundle.patients = bundle.patients.concat(newBundle.patients);
+    newBundle.total = bundle.total;
+    return newBundle;
+  }
+};
+
+export const addPatientsToBundle = (bundle: Bundle) => {
+  /*
+   *  addPatientsToBundle transforms a bundle in a PatientBundle (generate Patient objects)
+   */
+
+  let response: PatientBundle = bundle as PatientBundle;
+  response.patients = response.entry.map((entry: any) => {
+    const patient: Patient = {
+      id: entry.resource.id,
+      birthDate: entry.resource.birthDate,
+      age:
+        entry.resource.birthDate && getAge(new Date(entry.resource.birthDate))
+    };
+    if (entry.resource.name) {
+      if (entry.resource.name[0].given)
+        patient.firstName = entry.resource.name[0].given.join(", ");
+      if (entry.resource.name[0].family)
+        patient.lastName = entry.resource.name[0].family;
+    }
+    if (entry.resource.identifier) {
+      patient.identifier = entry.resource.identifier
+        .map((e: any) => {
+          return e.value;
+        })
```

Comment:
Instead of a function with only a return, you can write
`.map((e: any) => e.value)`

Reply:
Done ! I have to get used to this practice :p